lib/withOnyx.d.ts

```diff
@@ -42,7 +42,7 @@ type EntryBaseMapping<TOnyxKey extends OnyxKey> = {
  */
 type BaseMappingKey<TComponentProps, TOnyxProps, TOnyxProp extends keyof TOnyxProps, TOnyxKey extends OnyxKey, TOnyxValue> = IsEqual<TOnyxValue, TOnyxProps[TOnyxProp]> extends true
     ? {
-          key: TOnyxKey | ((props: Omit<TComponentProps, keyof TOnyxProps>) => TOnyxKey);
+          key: TOnyxKey | ((props: Omit<TComponentProps, keyof TOnyxProps> & Partial<TOnyxProps>) => TOnyxKey);
```

Comment:
Why is this better (or different) than `TComponentProps & TOnyxProps`? It appears to me like this is omitting properties, and then adding them right back with an intersection.

Reply:
We are doing two things here:
1. First, we omit the `TOnyxProps` from `TComponentProps` props because, when using the HOC, we pass `TComponentProps` and `TOnyxProps` separately, but `TComponentProps` always have all the props from `TOnyxProps`.
2. Now with that, we make a union with `Partial<TOnyxProps>` to make the Onyx props optional, because as @blazejkustra correctly pointed out the Onyx props are not available in this `key` function during the first render.

We can't do `TComponentProps & Partial<TOnyxProps>` directly because as `TComponentProps` has `TOnyxProps` props, making a union with `Partial<TOnyxProps>` won't make them optional.

In fact, if it weren't by the optional Onyx props, we could just do `TComponentProps` because `TComponentProps` always has all Onyx props from `TOnyxProps`. Here is an [example](https://github.com/Expensify/App/blob/main/src/components/ArchivedReportFooter.tsx#L23).

Reply:
Thanks for the explanation!